### PR TITLE
Create inital support for brms models

### PR DESCRIPTION
**Pull request by mutlusun**
# Description

This PR aims at adding support for brms models. This was relatively easy because most of the code can be used from stanreg.

# Proposed Changes

I added a file `report.brmsfit.R` for brms support.

# Remarks

- report_priors: I decided to deviate from the stanreg implementation. Unspecified priors in brms report a scale of `NA`. Thus, I treat models which only have priors with `scale == NA` as unspecified. I don't know exactly how this is handled in stanreg.
- Multivariate (multiple dependent variables) are not supported yet.
- Tests and examples with brms take much longer than any other tests in this package. Maybe you want to exclude them by default?
- I added also a test for stanreg. This was mainly to familiarize myself with the code. I am sorry that this is intermixed with the current PR.

I'm happy over any feedback!

Actually, I was a bit surprised how easy it is to extend the functionality of this package. I think you have created a really thought out API :)

### CODE REVIEW

File: R/report.stanreg.R [hunk 1]
```diff
@@ -18,7 +18,7 @@
 #'   as.data.frame(r)
 #'   summary(as.data.frame(r))
 #' }
-#' @include report.lm.R
+#' @include report.lm.R report.lme4.R
```
Comment (DominiqueMakowski) [inline R/report.stanreg.R:21]: I'm not sure that's necessary? 🤔

**Comment by DominiqueMakowski** [comment]
@mutlusun great thanks a lot! (Feel free to add yourself as contributor `role = cbt` in the [description](https://github.com/easystats/report/blob/master/DESCRIPTION) file :) I'll merge whenever you give me the green light

File: R/report.stanreg.R (hunk 1 above)
Comment (mutlusun) [reply R/report.stanreg.R:21]: But `report_random.merMod` is in `report.lme4.R` which is used in `report_random.stanreg`, isn't it?

**Comment by mutlusun** [comment]
Dear @DominiqueMakowski ,

Thanks for your feedback and adding me to the contributor list ;)

I think this can be merged but in the first message I forgot that following warnings occurred:

```
1: Can't extract 'deviance' residuals. Returning response residuals. 
2: Can't calculate weighted residuals from model. Model doesn't seem to have weights. 
3: Can't extract 'deviance' residuals. Returning response residuals. 
4: Can't calculate weighted residuals from model. Model doesn't seem to have weights. 
```

I have the latest `insight` from git and I don't know why these messages are thrown. The function to extract residuals is able to handle brms models.

**Comment by DominiqueMakowski** [comment]
@mutlusun you might want yourself as "ctb" in the [description](https://github.com/easystats/report/blob/master/DESCRIPTION) before I merge this :)

**Comment by DominiqueMakowski** [comment]
@strengejacke I was referring to here, for some reason it fails with:

```
The downloaded binary packages are in
	/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpCkk8an/downloaded_packages
Error in utils::download.file(url, path, method = method, quiet = quiet,  : 
  cannot open URL 'https://api.github.com/repos/easystats/insight/contents/DESCRIPTION?ref=HEAD'
Calls: saveRDS ... github_DESCRIPTION -> download -> base_download -> base_download_headers
Execution halted
Error: Process completed with exit code 1.
```

any idea?

**Comment by strengejacke** [comment]
No, not really. It looks like this issue (dis-)appears randomly.

**Pull request merged**